bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php

```diff
@@ -54,30 +54,141 @@ public function __construct(
+
+            // Bail out if there is nothing to bill for task.
+            if (0 === \count($worklogs) && 0 === \count($expenses)) {
+                continue;
+            }
 
-            // Add summary and task key to start of description.
+            $header->description = $header->description.(\strlen($header->description) > 0 ? ', ' : '').$task->fields->reporter->displayName.': '.$task->key;
+
+            $lines = [];
+
+            // Create line data for worklogs.
+            if (\count($worklogs) > 0) {
+                $worklogsSum = array_reduce($worklogs, function ($carry, $item) {
+                    $carry = $carry + $item->timeSpentSeconds;
+
+                    return $carry;
+                }, 0);
+
+                // From seconds to hours.
+                $worklogsSum = $worklogsSum / 60.0 / 60.0;
+
+                $lines[] = (object) [
+                    'materialNumber' => $this->boundMaterialId,
+                    'product' => 'Design: '.$library,
+                    'amount' => 1,
+                    'price' => $worklogsSum * $this->boundWorklogPricePerHour,
+                    'psp' => $this->boundReceiverPSP,
+                ];
+            }
+
+            // Create line data for expenses.
+            if (\count($expenses) > 0) {
+                $expensesSum = array_reduce($expenses, function ($carry, $item) {
+                    $carry = $carry + $item->amount;
+
+                    return $carry;
+                }, 0);
+
+                $lines[] = (object) [
+                    'materialNumber' => $this->boundMaterialId,
+                    'product' => 'Tryk: '.$library,
+                    'amount' => 1,
+                    'price' => $expensesSum,
+                    'psp' => $this->boundReceiverPSP,
+                ];
+            }
+
+            if (isset($entries[$library])) {
+                $entries[$library]->lines = array_merge($entries[$library]->lines, $lines);
+            } else {
+                $entries[$library] = (object) [
+                    'header' => $header,
+                    'lines' => $lines,
+                ];
+            }
+        }
+
+        return $entries;
+    }
+
+    /**
+     * Create export data for the given tasks.
+     *
+     * @param $tasks
+     *
+     * @return array
+     *
+     * @throws \Billing\Exception\InvoiceException
+     */
+    public function createExportDataNotMarketing($tasks)
+    {
+        // Get debtor custom field.
+        $debtorFieldId = $this->billingService->getCustomFieldId('Debitor');
+        $orderLinesFieldId = $this->billingService->getCustomFieldId('Order lines');
+        $entries = [];
+
+        foreach ($tasks as $task) {
             $description = implode('', [
-                $task->fields->summary,
-                ' ('.$task->key.'): ',
-                $description,
+                $task->key.': ',
+                preg_replace('/\\\\\\\\/', '.', $task->fields->{$orderLinesFieldId}),
```

Comment:
A comment about what's going on here would be nice. Consider adding a `getTaskDescription` method to handle this.

Reply:
Yes.